File: backend/src/main/java/com/bnorm/barkeep/server/data/store/v1/Book.java
package com.bnorm.barkeep.server.data.store.v1;

import java.util.ArrayList;
import java.util.List;

import com.bnorm.barkeep.server.data.store.Deref;
import com.googlecode.objectify.Ref;
import com.googlecode.objectify.annotation.Entity;
import com.googlecode.objectify.annotation.Id;

@Entity
public class Book {

    @Id private String name;
    private List<Ref<Recipe>> recipes = new ArrayList<>();

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public List<Recipe> getRecipes() {
        return Deref.deref(recipes);
    }

    public void setRecipes(List<Recipe> recipes) {
        this.recipes = Deref.ref(recipes);
    }
}

File: app/src/main/kotlin/com/bnorm/barkeep/data/api/model/Book.kt
package com.bnorm.barkeep.data.api.model

import android.os.Parcel
import android.os.Parcelable
import java.util.*

data class Book(var name: String? = null,
                var recipes: List<Recipe> = ArrayList()) : Parcelable {

    companion object {
        @JvmField val CREATOR = createParcel { Book(it) }
    }

    protected constructor(parcel: Parcel) :
            this(parcel.readString(),
                    parcel.createTypedArrayList(Recipe.CREATOR))

    override fun describeContents(): Int {
        return 0
    }

    override fun writeToParcel(dest: Parcel, flags: Int) {
        dest.writeString(name)
        dest.writeTypedList(recipes)
    }
}
